Lib/test/test_tools/test_i18n/test_i18n.py

```diff
@@ -310,6 +337,24 @@ def test_calls_in_fstring_with_partially_wrong_expression(self):
         self.assertNotIn('foo', msgids)
         self.assertIn('bar', msgids)
 
+    def test_pygettext_output(self):
+        """Test that the pygettext output exactly matches a file."""
+        data_dir = Path(__file__).parent / 'data'
```

Comment:
Perhaps define this at the module level?

```suggestion
        data_dir = Path(__file__).resolve().parent / 'data'
```